CosmoTech_Acceleration_Library/Accelerators/scenario_download/scenario_downloader.py

```diff
@@ -290,15 +340,17 @@ def download_dataset_process(_dataset_id, _return_dict, _error_dict):
             return_dict = manager.dict()
             error_dict = manager.dict()
             processes = [
-                (dataset_id, multiprocessing.Process(target=download_dataset_process, args=(dataset_id, return_dict, error_dict)))
+                (dataset_id, multiprocessing.Process(target=download_dataset_process,
+                                                     args=(dataset_id, return_dict, error_dict)))
                 for dataset_id in dataset_ids
             ]
             [p.start() for _, p in processes]
             [p.join() for _, p in processes]
+
             for dataset_id, p in processes:
-                if p.exitcode != 0:
+                if p.exitcode != 0 and error_dict.get(dataset_id) is not None:
```

Comment:
I think `dataset_id in error_dict` would be more pythonic to check if we have an error message for a specific dataset.

I would also document that this check is only required as a bug workaround. Something like:
```python
# We might hit the following bug: https://bugs.python.org/issue43944
# As a workaround, only treat non-null exit code as a real issue if we also have stored an error message
```